### PR TITLE
ci: add ability to have replaces,skips in CSV

### PR DESCRIPTION
**Pull request by Rakshith-R**
Each CSV has a replaces parameter that indicates
which Operator it replaces. This builds a graph of
CSVs that can be queried by OLM, and updates can be
shared between channels. Channels can be thought of
as entry points into the graph of updates:
REPLACES ?= ""

Creating the New CatalogSource requires publishing CSVs
that replace one Operator, but can skip several. This can
be accomplished using the skipRange annotation:
SKIP_RANGE ?= ""

Signed-off-by: Rakshith R <rar@redhat.com>

Refer: https://github.com/red-hat-storage/odf-operator/commit/e1c41bee0b26731a923eb06f5fc3e047539bfd6a

cc @nixpanic @Madhu-1 @iamniting

### CODE REVIEW

**Comment by iamniting** [comment]
@Rakshith-R Did you try to run the make bundle after feeding only one of the values? Does it work well?

**Comment by Rakshith-R** [comment]
> @Rakshith-R Did you try to run the make bundle after feeding only one of the values? Does it work well?

It fills in empty quotes `""` when no value is given.

File: Makefile [hunk 1]
```diff
@@ -90,6 +100,8 @@ manifests: controller-gen kustomize ## Generate WebhookConfiguration, ClusterRol
 # generate the <package-name>.clusterserviceversion.yaml
 config/manifests/bases/$(PACKAGE_NAME).clusterserviceversion.yaml: config/manifests/bases/clusterserviceversion.yaml.in
 	sed 's/@PACKAGE_NAME@/$(PACKAGE_NAME)/g' < $^ > $@
+	sed -i 's/@SKIP_RANGE@/$(SKIP_RANGE)/g' $@
```
Comment (Madhu-1) [inline Makefile:103]: `-i` is required?

Comment (Rakshith-R) [reply Makefile:103]: Yes, that tells sed to make the replace inplace

**Review by Madhu-1 — APPROVED** [review]

**Comment by iamniting** [comment]
> > @Rakshith-R Did you try to run the make bundle after feeding only one of the values? Does it work well?
> 
> It fills in empty quotes `""` when no value is given. No errors.

and what about the generated CSV, does it contain the "" quotes?

**Review by Madhu-1 — CHANGES_REQUESTED** [review]
Requesting changes to avoid accidental merge. as still discussion is going on to see as the keys need to be dropped for empty values.

**Review by iamniting — APPROVED** [review]

**Review by Madhu-1 — DISMISSED** [review]

File: Makefile [hunk 2]
```diff
@@ -87,12 +97,13 @@ manifests: controller-gen kustomize ## Generate WebhookConfiguration, ClusterRol
 	$(KUSTOMIZE) build config/rbac > deploy/controller/rbac.yaml
 	$(KUSTOMIZE) build config/manager > deploy/controller/setup-controller.yaml
 
-# generate the <package-name>.clusterserviceversion.yaml
-config/manifests/bases/$(PACKAGE_NAME).clusterserviceversion.yaml: config/manifests/bases/clusterserviceversion.yaml.in
-	sed 's/@PACKAGE_NAME@/$(PACKAGE_NAME)/g' < $^ > $@
+# generate the <package-name>.clusterserviceversion.yaml base
+gen-csv-base:
+	sed 's/@PACKAGE_NAME@/$(PACKAGE_NAME)/g;s/@SKIP_RANGE@/$(SKIP_RANGE)/g;s/@REPLACES@/$(REPLACES)/g' \
+	< config/manifests/bases/clusterserviceversion.yaml.in > config/manifests/bases/$(PACKAGE_NAME).clusterserviceversion.yaml
```
Comment (Rakshith-R) [inline Makefile:103]: Previously, changing replace and skip_range variables did not lead to creation of new base csv yaml. 

Now, new file is generated each time.

**Review by yati1998 — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**